### ISSUE TITLE
Improve DHIS2 automation

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-core#7469]
**Is your feature request related to a problem? Please describe.**
It is currently a manual process to get DHIS2 integration to work on a regular basis. It looks something like this:
1. Hit a URL in the CHT to download the data.
1. Log into DHIS2 in your browser.
1. Upload the data from step 1 into DHIS2 via an import screen.

**Describe the solution you'd like**
Automate the above steps so no humans need to be involved for reliable import into DHIS2.

**Describe alternatives you've considered**
The is a "no dev" solution that exists [using Apache NiFi](https://drive.google.com/file/d/1lyTNIR2KfnVXIY60bBweABCP5R67Fay1/view), but a CHT native solution would be much better.

**Additional context**
A number of partners have requested DHIS2 integration and if it's not automated it will slow or prevent adoption.

**Comment by jdndiaye** [comment]
Our feature request is not related to a problem. The manual process you described works well, the partners would like it to be automated in the future so that no one needs to be involved for a reliable import into DHIS2. This isn't an urgent need, for now, they are fine with going with the manual process.

**Comment by mrjones-plip** [comment]
Thanks for the update @jdndiaye! 

@craig-landry  - since this is not a pressing feature, I'm going to remove it [from 3.15](https://github.com/medic/cht-core/milestone/98) in hopes of keeping a lean release. If this isn't right, bring it back or ping me on slack so we can discuss!

**Comment by jdndiaye** [comment]
Hi @mrjones-plip 
Sorry, I spoke with Fatou and the MOH information systems manager. And it is clear that automation is now becoming a priority in this project. Please take this into account in the first phase.

Best,

**Comment by mrjones-plip** [comment]
Thanks for the update @jdndiaye ! Let's put it back in 3.15 then :rocket:

**Comment by mrjones-plip** [comment]
Noting that there's now the [App Services effort](https://github.com/medic/app-services-team/tree/master/dhis2-from-datasource) which, after configuration, is a single CLI call which  automates pulling results directly from postgres in materialized views to DHIS.

cc @garethbowen  for any architectural concerns and duplication of efforts

**Comment by magp18** [comment]
> Noting that there's now the [App Services effort](https://github.com/medic/app-services-team/tree/master/dhis2-from-datasource) which, after configuration, is a single CLI call which automates pulling results directly from postgres in materialized views to DHIS.
> 
> cc @garethbowen for any architectural concerns and duplication of efforts

The link seems to be broken.

**Comment by mrjones-plip** [comment]
Hey @magp18 ! Thanks for your comment.

Sorry for the bad link to the "App services effort" - GitHub shows resources you don't have permissions for (like a private repo in this case) as "404" which isn't that helpful.  

Further, I'll see if I'm allowed to post a copy of the solution I linked to.  What was done on that link was a node app that would do an [ETL](https://en.wikipedia.org/wiki/Extract,_transform,_load). The difference from this ticket is that it leveraged and existing [couch2pg](https://github.com/medic/cht-couch2pg/) deployment which, in turn, had materialized views run in PostgreSQL. These views were close to what was needed in DHIS2, such that an export from PostgreSQL -> light transform -> import into DHIS2 is what was done in the node app.

Speaking of an ETL process - I don't know the details of how to write a DHIS2 app, but as long as it can reach the CHT instance, can authenticate and query the right APIs, do any transforms if needed, then yes - a DHIS2 app could be written!

**Comment by mrjones-plip** [comment]
Follow up - I got the OK to post [the code for the node app](https://github.com/medic/cht-core/files/8180031/dhis2-from-datasource.zip) that pulls from PostgreSQL -> DHIS2. The PostgreSQL data was seeded by couch2pg and then enhance with materialized views. cc @magp18 .

**Comment by kennsippell** [comment]
> Can this not be done by building a DHIS2 app that pulls data from CHT?

@magp18  Yes. It is very possible. But I believe this would make a "one click integration with CHT from within the DHIS2 experience" while this ticket tracks improvements to fully automate the integration (no clicks in either platform).  I'm not sure how to do this as a DHIS2 app but would be interested to learn.

**Comment by magp18** [comment]
Thanks @mrjones-plip and @kennsippell for the input and clarification. Will have a look at the material provided too. Yes I am not sure how to do this as a DHIS2 app but heard about the possibility so thought about looking into it.

**Comment by garethbowen** [comment]
Not a blocker for 3.15.0
